src/lib/service/handlers.ts

```diff
@@ -0,0 +1,108 @@
+import { CreateContext, DeleteContext, DetailContext, ListContext, UpdateContext } from 'lib/context/crudContext';
+import { Operation } from 'lib/context/operation';
+import { errorOrEmpty } from 'lib/helpers';
+import { ServiceImplementation } from 'lib/settings/definitions';
+
+export const createHandlers = <T extends { id: any }, C extends object>(
+    implementation: ServiceImplementation<T, C>
+) => {
+    /**
+     * Fetch resource, throw error when resource missing.
+     * This method is used for handlers working with a single existing resource (get, update, delete)
+     */
+    const getSafe = (context: Pick<DetailContext<T, C>, 'id' | 'context' | 'options'>): Promise<T> =>
```

Comment:
I'd change `Promise` - which accepts only genuine ES6 Promises to a more generic PromiseLike.

AFAIK Promise fails for e.g. Bluebird Promises (used by knex and most libraries)

Reply:
Good point, did not know. I like that. Fixed.